sdk/cosmosdb/cosmos/test/public/integration/changeFeedIterator.spec.ts

```diff
@@ -435,3 +442,461 @@ describe("test changefeed for feed range", function () {
     assert.equal(counter, 40, "40 items should be fetched");
   });
 });
+
+describe("test changefeed allVersionsAndDeletes mode for entire container", function () {
```

Comment:
Can you test when containers have split partitions?

Reply:
This is already a multi-partitioned container. Are you asking to simulate a partition split in the code?

Reply:
The Java and .NET SDK supports split containers. I am not sure if that is something that you have thought about and I can't remember if we discussed this on a call or not. But split containers are supported if the CX is not in direct mode, but gateway mode, because that is where the split handling logic lives. I am open for a conversation about this because I think that it is an important discussion to have.

Reply:
JS SDK only supports gateway mode. We are handling a partition split by checking the error code we get(410) from backend and subsequently enqueuing children partitions in the internal queue.